Namespace Juicebox V1 contract constants

aeolianeth opened the pull request:
## What does this PR do and why?

- Move `constants/contracts.ts` to `constants/v1/contracts.ts`
- Move `ContractName` typedef to `constants/v1/contracts.ts`. Delete `contract-name.ts` file.
- Rename `ContractName` type to `V1ContractName`
- Rename `Contracts` type to `V1Contracts`

| iteration | PR |
| --- | --- |
| 1. Namespace Juicebox V1 contract constants | 👉 you are here |
| 2. Namespace Juicebox V1 terminal code | https://github.com/jbx-protocol/juice-interface/pull/436 |
| 3. Namespace contexts and providers | https://github.com/jbx-protocol/juice-interface/pull/438 |

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [-] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

aeolianeth commented:
First one for you @peripheralist 🏀

peripheralist commented:
let's gooo! @aeolianeth think we could get away with using a more terse `V1ContractName` etc, instead of `JuiceboxV1ContractName`? "Juicebox" feels kinda implicit

aeolianeth commented:
> let's gooo! @aeolianeth think we could get away with using a more terse `V1ContractName` etc, instead of `JuiceboxV1ContractName`? "Juicebox" feels kinda implicit (also applies to #436 )

@peripheralist I don't feel strongly! A simple `V1` prefix is super repeatable as well for things like UserContext (we just change it to `V1UserContext`).

Changed and updated #436 and #438 🚀

peripheralist commented on src/hooks/ContractLoader.ts:
```diff
@@ -48,7 +48,7 @@ export function useContractLoader() {
 
 const loadContract = (
-  contractName: string,
+  contractName: keyof typeof V1ContractName,
```
👏

peripheralist commented:
love this @aeolianeth !
